extensions/vscode/src/multiStepInputs/newDeployment.ts

```diff
@@ -400,9 +407,12 @@ export async function newDeployment(
 
   const getContentRecords = new Promise<void>(async (resolve, reject) => {
     try {
-      const response = await api.contentRecords.getAll(".", {
-        recursive: true,
-      });
+      const response = await api.contentRecords.getAll(
+        projectDir ? projectDir : ".",
+        {
+          recursive: true,
+        },
```

Comment:
Does this need to run recursively when there's a project dir?

Reply:
Yes, indeed. Good catch. Fix pushed up.

Reply:
Double checking this since it is still `projectDir ? false : true`

We don't actually want to be recursive when a projectDir, the same as `getConfigurationInspections` since the `projectDir` param was passed. That is done when the `showNewDeploymentMultiStep` gets a `projectDir` only when we initiate from a entrypoint file - so we don't want a recursion we only want the one entrypoint.

However PreDeployment records are not filtered on entrypoints so we have to do it manually here.

I just want to be sure I'm understanding that, and this code should stay the same?

Reply:
Yes, sorry, my bad. When there is no project dir, it needs to be run recursively. When there is a project dir passed in, there is no need for us to run recursively, because we won't have entrypoint selection across multiple dirs, which then selects which deployment names need to be considered.

The code has been updated ahead of you seeing it and it is correct.